app/src/main/assets/util_functions.sh

```diff
@@ -238,11 +238,21 @@ get_next_slot() {
 }
 
 find_boot_image() {
-  if [ ! -z $SLOT ]; then
-    BOOTIMAGE=$(find_block "boot$SLOT")
-  fi
-  if [ -z $BOOTIMAGE ]; then
-    BOOTIMAGE=$(find_block kern-a android_boot kernel bootimg boot lnx boot_a)
+  if $RECOVERYMODE; then
+    BOOTIMAGE=$(find_block "recovery$SLOT" "sos")
+  elif [ -L "/dev/block/by-name/init_boot$SLOT" ] && uname -r | grep -vq "android12-"; then
+    # init_boot is only used with GKI 13+. It is possible that some devices with init_boot
```

Comment:
Unlike Magisk, we should always patch kernel. But most of devices' kernel is in boot image instead of init_boot I thought. Maybe we need a discussion about this?